hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/util/PartitionFilterGenerator.java

```diff
@@ -0,0 +1,191 @@
+import static org.apache.hudi.sync.common.HoodieSyncConfig.META_SYNC_PARTITION_EXTRACTOR_CLASS;
+
+public class PartitionFilterGenerator {
+
+  /**
+   * Build expression from the Partition list.
+   *
+   * ex. partitionSchema(date, hour) [Partition(2022-09-01, 12), Partition(2022-09-02, 13)] =>
+   *     Or(And(Equal(Attribute(date), Literal(2022-09-01)), Equal(Attribute(hour), Literal(12))),
+   *     And(Equal(Attribute(date), Literal(2022-09-02)), Equal(Attribute(hour), Literal(13))))
+   */
+  private static Expression buildPartitionExpression(List<Partition> partitions, List<FieldSchema> partitionFields) {
+    return partitions.stream().map(partition -> {
+      List<String> partitionValues = partition.getValues();
+      Expression root = null;
+
+      for (int i = 0; i < partitionFields.size(); i++) {
+        FieldSchema field = partitionFields.get(i);
+        String value = partitionValues.get(i);
+        BinaryOperator.EqualTo exp = new BinaryOperator.EqualTo(new LeafExpression.AttributeReferenceExpression(field.getName()),
+            new LeafExpression.Literal(value, field.getType()));
+        if (root != null) {
+          root = new BinaryOperator.And(root, exp);
+        } else {
+          root = exp;
+        }
+      }
+      return root;
+    }).reduce(null, (result, expr) -> {
+      if (result == null) {
+        return expr;
+      } else {
+        return new BinaryOperator.Or(result, expr);
+      }
+    });
+  }
+
+  /**
+   * Extract partition values from the {@param partitions}, and binding to
+   * corresponding partition fieldSchemas.
+   */
+  private static List<Pair<FieldSchema, String[]>> extractFieldValues(List<Partition> partitions, List<FieldSchema> partitionFields) {
+    return IntStream.range(0, partitionFields.size())
+        .mapToObj(i -> {
+          Set<String> values = new HashSet<String>();
+          for (int j = 0; j < partitions.size(); j++) {
+            values.add(partitions.get(j).getValues().get(i));
+          }
+          return Pair.of(partitionFields.get(i), values.toArray(new String[0]));
+        })
+        .collect(Collectors.toList());
+  }
+
+  private static class ValueComparator implements Comparator<String> {
+
+    private final String valueType;
+    public ValueComparator(String type) {
+      this.valueType = type;
+    }
+
+    @Override
+    public int compare(String s1, String s2) {
+      switch (valueType.toLowerCase(Locale.ROOT)) {
+        case HiveSchemaUtil.INT_TYPE_NAME:
+          int i1 = Integer.parseInt(s1);
+          int i2 = Integer.parseInt(s2);
+          return i1 - i2;
+        case HiveSchemaUtil.BIGINT_TYPE_NAME:
+          long l1 = Long.parseLong(s1);
+          long l2 = Long.parseLong(s2);
+          long result = l1 - l2;
+          if (result > 0) {
+            return 1;
+          }
+
+          if (result < 0) {
+            return -1;
+          }
+
+          return 0;
+        default:
+          return s1.compareTo(s2);
```

Comment:
What about other types proper comparison? We can't compare booleans, floats, etc as strings either

Reply:
Make sense. Please add it as a comment in the code as well, so that whoever reading this won't be guessing